kjdoc/tour.md

```diff
@@ -538,13 +538,441 @@ This section describes KJ APIs that control process execution and low-level inte
 
 ## System I/O
 
-TODO: async-io, etc.
+TODO: OS handles, sync IO, async IO, filesystem, network, time
+
+### Async I/O
+
+On top of KJ's async framework (described earlier), KJ provides asynchronous APIs for byte streams, networking, and timers.
+
+As mentioned previously, `kj::setupAsyncIo()` allocates an appropriate OS-specific event queue (such as `epoll` on Linux), returning implementations of `kj::AsyncIoProvider` and `kj::LowLevelAsyncIoProvider` implemented in terms of that queue. `kj::AsyncIoProvider` provides an OS-independent API for byte streams, networking, and timers. `kj::LowLevelAsyncIoProvider` allows native OS handles (file descriptors on Unix, `HANDLE`s on Windows) to be wrapped in KJ byte stream APIs, like `kj::AsyncIoStream`.
+
+Please refer to the API reference (the header files) for details on these APIs.
+
+### Synchronous I/O
+
+Although most complex KJ applications use Async I/O, sometimes you want something a little simpler.
+
+`kj/io.h` provides some more basic, synchronous streaming interfaces, like `kj::InputStream` and `kj::OutputStream`. Implementations are provided on top of file descriptors and Windows `HANDLE`s.
+
+Additionally, the important utility class `kj::AutoCloseFd` (and `kj::AutoCloseHandle` for Windows) can be found here. This is an RAII wrapper around a file descriptor (or `HANDLE`), which you will likely want to use any time you are manipulating raw file descriptors (or `HANDLE`s) in KJ code.
+
+### Filesystem
+
+KJ provides an advanced, cross-platform filesystem API in `kj/filesystem.h`. Features include:
+
+* Paths represented using `kj::Path`. In addition to providing common-sense path parsing and manipulation functions, this class is designed to defend against path injection attacks.
+* All interfaces are abstract, allowing mulitple implementations.
+* An in-memory implementation is provided, useful in particular for mocking the filesystem in unit tests.
+* On Unix, disk `kj::Directory` objects are backed by open file descriptors and use the `openat()` family of system calls.
+* Makes it easy to use atomic replacement when writing new files -- and even whole directories.
+* Symlinks, hard links, listing directories, recursive delete, recursive create parents, recursive copy directory, memory mapping, and unnamed temporary files are all exposed and easy to use.
+* Sparse files ("hole punching"), copy-on-write file cloning (`FICLONE`, `FICLONERANGE`), `sendfile()`-based copying, `renameat2()` atomic replacements, and more will automatically be used when available.
+
+See the API reference (header file) for details.
+
+### Clocks and time
+
+KJ provides a time library in `kj/time.h` whicth uses the type system to enforce unit safety.
```

Comment:
Misspelling of which

Reply:
```suggestion
KJ provides a time library in `kj/time.h` which uses the type system to enforce unit safety.
```

@Dr-Emann you can use the button to the left of the style buttons to add a suggestion, which makes it super easy for the PR author to incorporate the change. It's the button highlighted in blue in this screenshot:

![Screenshot from 2021-01-22 09-50-34](https://user-images.githubusercontent.com/1030771/105487068-60799d80-5c64-11eb-84d5-e378cfe26cc9.png)